modules/cleanmedianetBidAdapter.js

```diff
@@ -0,0 +1,297 @@
+      ) {
+        rtbBidRequest.ext.gdpr_consent = {
+          consent_string: bidderRequest.gdprConsent.consentString,
+          consent_required: bidderRequest.gdprConsent.gdprApplies
+        };
+      }
+
+      const imp = {
+        id: transactionId,
+        instl: params.instl === 1 ? 1 : 0,
+        tagid: adUnitCode,
+        bidfloor: params.bidfloor || 0,
+        bidfloorcur: 'USD',
+        secure: 1
+      };
+
+      const hasFavoredMediaType =
+        params.favoredMediaType &&
+        this.supportedMediaTypes.includes(params.favoredMediaType);
+
+      if (!mediaTypes || mediaTypes.banner) {
+        if (!hasFavoredMediaType || params.favoredMediaType === BANNER) {
+          const bannerImp = Object.assign({}, imp, {
+            banner: {
+              w: sizes.length ? sizes[0][0] : 300,
+              h: sizes.length ? sizes[0][1] : 250,
+              pos: params.pos || 0,
+              topframe: bidderRequest.refererInfo.reachedTop
+            }
+          });
+          rtbBidRequest.imp.push(bannerImp);
+        }
+      }
+
+      if (mediaTypes && mediaTypes.video) {
+        if (!hasFavoredMediaType || params.favoredMediaType === VIDEO) {
+          let videoImp = {
+            video: {
+              protocols: params.protocols || [1, 2, 3, 4, 5, 6],
+              pos: params.pos || 0,
+              ext: {context: mediaTypes.video.context}
+            }
+          };
+
+          let playerSize = mediaTypes.video.playerSize || sizes;
+          if (utils.isArray(playerSize[0])) {
+            videoImp.video.w = playerSize[0][0];
+            videoImp.video.h = playerSize[0][1];
+          } else if (utils.isNumber(playerSize[0])) {
+            videoImp.video.w = playerSize[0];
+            videoImp.video.h = playerSize[1];
+          } else {
+            videoImp.video.w = 300;
+            videoImp.video.h = 250;
+          }
+
+          videoImp = Object.assign({}, imp, videoImp);
+          rtbBidRequest.imp.push(videoImp);
+        }
+      }
+
+      if (rtbBidRequest.imp.length === 0) {
+        return;
+      }
+
+      return {
+        method: 'POST',
+        url: rtbEndpoint,
+        data: rtbBidRequest,
+        bidRequest
+      };
+    });
+  },
+
+  interpretResponse: function (serverResponse, bidRequest) {
+    const response = serverResponse && serverResponse.body;
+    if (!response) {
+      utils.logError('empty response');
+      return [];
+    }
+
+    const bids = response.seatbid.reduce(
+      (acc, seatBid) => acc.concat(seatBid.bid),
+      []
+    );
+    let outBids = [];
+
+    bids.forEach(bid => {
+      const outBid = {
+        adId: bidRequest.bidRequest.adUnitCode,
+        requestId: bidRequest.bidRequest.bidId,
+        cpm: bid.price,
+        width: bid.w,
+        height: bid.h,
+        ttl: 60 * 10,
+        creativeId: bid.crid,
+        netRevenue: true,
+        currency: bid.cur || response.cur,
+        adUnitCode: bidRequest.bidRequest.adUnitCode,
```

Comment:
This is unnecessary in my opinion because prebid will already assign the correct bidResponse to the correct ad unit and thus adUnitCode based on the requestId you pass in (Which should match to the correct bidId)

What are the reasonings for adding this adUnitCode in these places?

Reply:
please see my comment above